console/src/components/Services/RemoteSchema/Permissions/Container.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { ReactElement } from 'react';
+import CommonTabLayout from '../../../Common/Layout/CommonTabLayout/CommonTabLayout';
+import { NotFoundError } from '../../../Error/PageNotFound';
+import { appPrefix } from '../constants';
+import styles from '../RemoteSchema.scss';
+import { RSPContainerProps } from './types';
+
+const tabInfo = {
+  details: {
+    display_text: 'Details',
+  },
+  modify: {
+    display_text: 'Modify',
+  },
+  permissions: {
+    display_text: 'Permissions',
+  },
+};
+
+const RSPContainer: React.FC<RSPContainerProps> = ({
+  params: { remoteSchemaName },
+  allRemoteSchemas,
+  tabName,
+  viewRemoteSchema,
+  children,
+}) => {
+  React.useEffect(() => {
+    viewRemoteSchema(remoteSchemaName);
+    return () => {
+      viewRemoteSchema('');
+    };
+  }, [remoteSchemaName]);
+
+  const currentRemoteSchema = allRemoteSchemas.find(
+    rs => rs.name === remoteSchemaName
+  );
+
+  if (!currentRemoteSchema) {
+    viewRemoteSchema('');
+    throw new NotFoundError();
+  }
+
+  const breadCrumbs = [
+    {
+      title: 'Remote schemas',
+      url: appPrefix,
+    },
+    {
+      title: 'Manage',
+      url: `${appPrefix}/manage`,
+    },
+    {
+      title: remoteSchemaName,
+      url: `${appPrefix}/manage/${remoteSchemaName}/modify`,
+    },
+    {
+      title: tabName,
+      url: '',
+    },
+  ];
+
+  const childrenWithProps = React.Children.map(children, child =>
+    React.cloneElement(child as ReactElement<any, any>, { currentRemoteSchema })
+  );
```

Comment:
Please don't use `cloneElement`. Consider using context or render props over the `cloneElement`.

|   | Redner props  | React.cloneElement  | React.Context  |
|---|---|---|---|
| Explicit data passing | ✅| ❌| ❌|
| Easy statically typed  | ✅| ❌| ✅|
| Flexible | ❌| ❌| ✅|